subsys/shell/shell.c

```diff
@@ -734,7 +734,7 @@ static int execute(const struct shell *shell)
 				      shell->ctx->cmd_buff,
 				      CONFIG_SHELL_ARGC_MAX);
 
-		if (shell->ctx->selected_cmd) {
+		if (IS_ENABLED(SHELL_CMDS_SELECT) && shell->ctx->selected_cmd) {
```

Comment:
Should be `CONFIG_SHELL_CMDS_SELECT` instead?

Reply:
yes! good catch